tests/test_modeling_common.py

```diff
@@ -1215,15 +1215,39 @@ def _create_and_check_torch_fx_tracing(self, config, inputs_dict, output_loss=Fa
                             (past_mask, inputs_to_test[1]["attention_mask"]), dim=1
                         )
 
-            if "inputs_embeds" in inspect.signature(model.forward).parameters and not model.config.is_encoder_decoder:
-                inputs_to_test.append(
-                    {
-                        "inputs_embeds": torch.rand(
-                            2, 2, model.config.hidden_size, dtype=torch.float, device=torch_device
-                        )
-                    }
+            if (
+                "input_ids" in inspect.signature(model.forward).parameters
+                and "inputs_embeds" in inspect.signature(model.forward).parameters
+                and not model.config.is_encoder_decoder
+            ):
```

Comment:
As far as I can tell this can just be combined with the block above as it also applies for when `not config.is_encoder_decoder`

We can then make this whole logic check cheaper and only inspect the params once 

```suggestion
                forward_parameters = inspect.signature(model.forward).parameters
                if "input_ids" in forward_parameters and "input_embeds" in forward_parameters):
```